### PR TITLE
362 beneficial owner city3

### PR DESCRIPTION
**Pull request by BenGalewsky**
# Summary
The beneficial owner section of the ownership information page on the enrollment workflow has
had some problems which were hard to pin down. During automated testing of the enrollment process it became clear that the city field for the individual beneficial owner's other medicaid provider division was not being accepted.

This PR contains a test that demonstrated this bug along with a fix to the javascript to fix the bug.

# Testing Approach
A new Serenity test feature was created that makes use of some new test steps to drive through the enrollment workflow. At the end of the workflow it creates a beneficial owner for the provider and clicks on the checkbox which states that the owner has a controlling interest in another medicaid provider. At that point it fills in name and address information on that other provider and clicks `next` to accept the values and progress to the next screen.

The test checks to make sure the data validates and the next screen is presented. 

# Fixing The Bug
The number of beneficial owners on an enrollment page is dynamic. When the user clicks to add a 
new beneficial owner, JavaScript runs behind the scenes to create new form fields. The naming of these fields is critical to binding the entered data to the database.

The field names are of the form `_17_cboOtherZip_XX` where XX is the index of this beneficial owner on the page, starting at zero.

There is a method in `cms-web/WebContent/js/script.js` that names all of the fields in the newly added sub-form. This method is called `reindexPersonOwners`.

Upon close inspection of this method I could see that several of the fields were not being correctly named or had been left out of this method. The city field just happened to be one of the mandatory fields so it triggered an error when the form was submitted.

This method is rather brittle since it assumes the order that the input and selects are added to the form and assigns names based on that index.

I opened up the DOM inspector console in Chrome while on this page to interrogate the fields in the subform to correctly light up the names assigned in this method.

# Validating the Fix
I rebuilt the EAR file with `./gradlew clean cms-web:clean cms-web:build cms-portal-services:build` and deployed to Wildfly. I ran the full suite of serenity tests and they now passed.

### CODE REVIEW

File: psm-app/cms-web/WebContent/js/script.js [hunk 1]
```diff
@@ -1778,13 +1778,16 @@ $(document).ready(function() {
...
-			$i.find('select:eq(2)').attr("name", "_17_iboOtherState_" + index);
-			$i.find('input:eq(22)').attr("name", "_17_iboOtherZip_" + index);
+			$i.find('select:eq(2)').attr("name", "_17_iboCounty_" + index);
+			$i.find('input:eq(17)').attr("name", "_17_iboOtherInterestInd_" + index);
+            $i.find('input:eq(18)').attr("name", "_17_iboOtherInterestName_" + index);
+            $i.find('input:eq(19)').attr("name", "_17_iboOtherInterestPct_" + index);
+            $i.find('input:eq(20)').attr("name", "_17_iboOtherAddressLine1_" + index);
+            $i.find('input:eq(21)').attr("name", "_17_iboOtherAddressLine2_" + index);
+            $i.find('input:eq(22)').attr("name", "_17_iboOtherCity_" + index);
+            $i.find('select:eq(3)').attr("name", "_17_iboOtherCounty" + index);
```
Comment (jasonaowen) [inline psm-app/cms-web/WebContent/js/script.js]: Did you mean for these lines to be indented differently? If not, could you please fix?

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        enrollmentPage.setOwnershipDOB("01011970");
+        enrollmentPage.setOwnershipHireDate("01012000");
+        enrollmentPage.setOwnershipCity("Ownertown");
+        enrollmentPage.selectOwnershipState("Texas");
+        enrollmentPage.setOwnershipZip("77706");
+        enrollmentPage.selectOwnershipCounty("Beltrami");
+    }
+
+    @Given("^I have indicated that the owner has an interest in another Medicaid disclosing entity$")
+    public void i_have_indicated_that_the_owner_has_an_interest_in_another_Medicaid_disclosing_entity() throws Throwable {
```
Comment (jasonaowen) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java]: No `throws Throwable`, please. :)

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/EnrollmentPage.java [hunk 3]
```diff
@@ -59,4 +63,191 @@ public boolean isPersonalEnrollment() {
...
+    }
+
+    public void selectControlOwnershipState(String state) {
+        $("[name='_17_iboOtherState_0']").selectByVisibleText(state);
+    }
+
+
+    public void selectControlOwnershipCounty(String county) {
+        $("[name='_17_iboOtherCounty0']").selectByVisibleText(county);
+    }
```
Comment (jasonaowen) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/EnrollmentPage.java]: This file is getting pretty big; would it make sense to split it into multiple `PageObject`s? Like, one for the screen where you choose a provider type, one for individual provider information, and so on?

**Comment by BenGalewsky** [comment]
@jasonaowen thanks for prodding me to refactor the enrollment PageObject class. It was well worth it.

**Pull request merged**